src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1240 @@
+             'cbsdId': resp['cbsdId'],
+             'grantId': resp['grantId']
+        })
+
+    # Send Consolidated Relinquishment Request for all the grants,
+    # if the grant responseCode was 501
+    if len(relq_requests['relinquishmentRequest']) != 0:
+        relq_responses = self._sas.Relinquishment(relq_requests)['relinquishmentResponse']
+        self.assertEqual(len(relq_responses), len(relq_requests['relinquishmentRequest']))
+        for relq_resp in relq_responses:
+          # Check the relinquishment response
+          self.assertEqual(relq_resp['response']['responseCode'], 0)
+
+        del relq_requests, relq_responses
+
+    del heartbeat_requests, heartbeat_responses
+
+    # Step 6: Request grant 'G2' with another frequency range which partially or fully
+    # overlaps with Exclusion zone protected frequency range 'F1'.
+    # Forming of 'G2' grants for all the configured grants
+    addCbsdIdsToRequests(cbsd_ids, config['grantRequests'][1])
+    grant_request_g2 = {'grantRequest': config['grantRequests'][1]}
+
+    # Send grant request 'G2'
+    grant_response_g2 = self._sas.Grant(grant_request_g2)['grantResponse']
+    self.assertEqual(len(grant_response_g2), len(grant_request_g2['grantRequest']))
+
+    # Check grant response,
+    # responseCode should be 400 (INTERFERENCE).
+    for resp in grant_response_g2:
+      self.assertEqual(resp['response']['responseCode'], 400)
+
+    del grant_request_g2, grant_response_g2
+
+    # TODO
+    # Step 7: Modify exclusion zone database record frequency range from 'F1' to 'F2'
+
+    # Step 8: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 9: Request grant 'G3' with frequency range which partially or fully
+    # overlaps with Exclusion zone protected frequency range 'F2'.
+    # Forming of 'G3' grants for all the configured grants
+    addCbsdIdsToRequests(cbsd_ids, config['grantRequests'][2])
+    grant_request_g3 = {'grantRequest': config['grantRequests'][2]}
+
+    # Send grant request 'G3'
+    grant_response_g3 = self._sas.Grant(grant_request_g3)['grantResponse']
+    self.assertEqual(len(grant_response_g3), len(grant_request_g3['grantRequest']))
+
+    # Check grant response,
+    # responseCode should be 400 (INTERFERENCE).
+    for resp in grant_response_g3:
+      self.assertEqual(resp['response']['responseCode'], 400)
+
+    del grant_request_g3, grant_response_g3
+
+  def generate_FDB_2_default_config(self, filename):
+    """Generates the WinnForum configuration for FDB.2"""
+
+    # Load devices info
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # Load grant requests
+    grant_g1_a = json.load(
+                open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    # Set the grant frequency to overlap with the DPA 'puerto_rico_ver2_dpa_4'.
```

Comment:
Is this an ESC-monitored or a portal-controlled DPA?

Reply:
FDB.2        DPA Database Update for DPAs which cannot be monitored by an ESC